save-cli/src/commonMain/kotlin/org/cqfn/save/cli/Config.kt

```diff
@@ -8,79 +8,124 @@ import org.cqfn.save.core.config.LanguageType
 import org.cqfn.save.core.config.ReportType
 import org.cqfn.save.core.config.ResultOutputType
 import org.cqfn.save.core.config.SaveConfig
+import org.cqfn.save.core.logging.logDebug
+import org.cqfn.save.core.logging.logError
+import org.cqfn.save.core.logging.logInfo
 
+import okio.FileSystem
+import okio.IOException
 import okio.Path.Companion.toPath
 
+import kotlinx.cli.AbstractSingleOption
 import kotlinx.cli.ArgParser
 import kotlinx.cli.ArgType
+import kotlinx.cli.MultipleOption
 import kotlinx.cli.default
 import kotlinx.cli.multiple
 import kotlinx.cli.required
 
+private fun <U> Map<String, String>.getAndParseOrElse(
+    key: String,
+    parse: String.() -> U,
+    default: () -> U) =
+        get(key)?.let(parse) ?: default()
+
 /**
  * @param args CLI args
  * @return an instance of [SaveConfig]
+ * @throws e
  */
 @Suppress("TOO_LONG_FUNCTION")
 fun createConfigFromArgs(args: Array<String>): SaveConfig {
-    val parser = ArgParser("save")
+    // The first iteration of parsing - get location of properties file. Ignore any other arguments.
+    val propertiesFileOptionFullName = "properties-file"
+    val propertiesFileOptionShortName = "prop"
+    val propertiesFileName = args.toList().zipWithNext()
+        .firstOrNull { it.first == "--$propertiesFileOptionFullName" || it.first == "-$propertiesFileOptionShortName" }
+        ?.second
+        ?: "save.properties"
+    logDebug("Using properties file $propertiesFileName")
+
+    val properties: Map<String, String> = try {
+        FileSystem.SYSTEM.read(propertiesFileName.toPath()) {
+            generateSequence { readUtf8Line() }.toList()
+        }
+            .associate { line ->
+                line.split("=", limit = 2).let {
+                    it.first() to it.last()
+                }
+            }
+    } catch (e: IOException) {
+        logError("Unable to read properties file $propertiesFileName: ${e.message}")
+        emptyMap()  // todo exit with exit code?
```

Comment:
yep, I think that you just should be doing `exitProcess`

Reply:
I'll do it in a separate PR, because code should be moved from commonMain to nativeMain